### DISCUSSION TITLE
Making simdb optional

### DISCUSSION THREAD

**Discussion by timsnyder** [sparcians/map#473] [Ideas]
I don't use simdb (yet). Not sure I will.  I'm facing an issue where it would be nice if simdb could be optionally included in the build of Sparta so that I don't have to worry about it's unused 3rd party dependencies causing linking errors later on at runtime.

I'm starting to look at what it would take to add a cmake option to enable/disable simdb being included in the sparta build.  I'm curious to know whether anyone else would like to see this upstreamed or if the team views simdb as an integral part of sparta that shouldn't be considered optional?

**Comment by klingaard** [comment]
I like that idea a lot.  Not sure how integrated it is, however.  I don't use it yet either but plan to this go around.

**Comment by timsnyder** [comment]
@colby-nyce, now that you're back looking at things again, I'd be curious if you have thoughts about this topic.

**Comment by klingaard** [comment]
@colby-nyce is actually looking at this right now for map_v3.  See #509 I'm thinking about making this the "first release" of map_v3 as it's probably the _least_ impactful idea we have in that list.

**Comment by klingaard** [comment]
Related PR: #559

**Comment by colby-nyce** [comment]
Some thoughts on this... today, SimDB could be disabled pretty easily since it is not being used for much that we don't already have another solution for (reports, pipeline collection). But SimDB in map_v3 will be used for these important features, and we are not planning on maintaining and supporting both workflows and code bases (legacy reports + SimDB reports for instance).

So disabling SimDB will be the same as disabling all data collection. In that scenario, what will you be using Sparta for? I think end users care enough about reports (to say nothing of Argos) to make SimDB in v3 a fairly hard requirement for Sparta. 

If it helps, the new SimDB has been rewritten to be header-only and only require linking against sqlite3 and hdf5 (which are in SPARTA_LIBS).

All that being said: I'm happy to put in guardrails that will still conditionally compile the SimDB code in the presence of sqlite3/hdf5, and issue a warning/error if you try to use --report or -z without these libraries available.

**Reply by klingaard** [reply]
I think the main issue with SimDB is the use/need for additional software package like sqlite3, hdf5, etc.  

Agreed that once we move all reporting (v3) over to SimDB, it will be tightly coupled again; and we'll need folks to install those additional packages.  However, I do have a requirement that the SimDB submodule be part of the sparcians main project and not have it point to a user's side development repo.  It can be a fork of your simdb repo, but I do want it to reside here...
